src/java/org/apache/cassandra/io/sstable/StorageHandler.java

```diff
@@ -146,6 +151,16 @@ public StorageHandler(TableMetadataRef metadata, Directories directories, Tracke
      */
     public abstract void unload();
 
+    public abstract SSTableReader onOpeningWrittenSSTableFailure(SSTableReader.OpenReason reason,
```

Comment:
would it make sense to add a default implementation that throws the exception ? this way  when we upgrade Cassandra we don't break the build ?

this patch is a no-op outside of CNDB